Simplify MiniBrowser settings menu construction

hortont424 opened the pull request:
#### d8123acbf6c2d2ce59a59c6098c6dbf0f8201939
<pre>
Simplify MiniBrowser settings menu construction
<a href="https://bugs.webkit.org/show_bug.cgi?id=247305">https://bugs.webkit.org/show_bug.cgi?id=247305</a>

Reviewed by Simon Fraser.

* Tools/MiniBrowser/mac/SettingsController.m:
(-[SettingsController initWithMenu:]):
(addItemToMenuWithTarget):
(addSeparatorToMenu):
(addSubmenuToMenu):
(-[SettingsController _populateMenu:]):
(-[SettingsController buildUserAgentsMenu:]):
(-[SettingsController validateMenuItem:]):
(-[SettingsController changeCustomUserAgent:]):
(-[SettingsController dealloc]): Deleted.
(-[SettingsController _addItemWithTitle:action:indented:]): Deleted.
(-[SettingsController _addHeaderWithTitle:]): Deleted.
(-[SettingsController _populateMenu]): Deleted.
(-[SettingsController changeCutomUserAgent:]): Deleted.
Simplify the menu construction with some more helper functions.
Move the resource usage overlays into the overlay submenu.
Remove the empty WebKit1-only settings section.
Fix some typos.

Canonical link: <a href="https://commits.webkit.org/256195@main">https://commits.webkit.org/256195@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9dc15e87fe959543c34d4d2a09fdde79965c5e79

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/95023 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/4166 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/27927 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/104625 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/164878 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/99013 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/4254 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/32348 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/87329 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/100526 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/100690 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/3087 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/81542 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/30067 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/85001 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/84557 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/72963 "Found 3 new API test failures: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/state-changed, /TestWebKit:WebKit.NewFirstVisuallyNonEmptyLayoutFrames, /WebKitGTK/TestWebViewEditor:/webkit/WebKitWebView/cut-copy-paste/non-editable (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/38756 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/18382 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/36579 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/19663 "Found 30 new test failures: css3/color-filters/color-filter-animation.html, fast/dom/Window/a-rel-noopener.html, fast/events/popup-allowed-from-gesture-initiated-form-submit.html, fast/forms/submit-to-blank-multiple-times.html, http/tests/appcache/fail-on-update-2.html, http/tests/cache/disk-cache/resource-becomes-uncacheable.html, http/tests/cache/disk-cache/shattered-deduplication.html, http/tests/navigation/target-blank-opener-post.html, http/tests/privateClickMeasurement/multiple-app-bundle-ids.html, http/tests/privateClickMeasurement/private-click-measurement-with-source-nonce-null-content.html ... (failure)") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4277 "Found 1 new test failure: imported/w3c/web-platform-tests/html/semantics/links/links-created-by-a-and-area-elements/target_blank_implicit_noopener_base.html (failure)") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/40512 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/42359 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2052 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/42491 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/38898 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Tools/MiniBrowser/mac/SettingsController.m:
```diff
@@ -113,154 +107,120 @@ - (instancetype)initWithMenu:(NSMenu *)menu
 {
-    [_menu addItem:[NSMenuItem separatorItem]];
-    [_menu addItem:[[[NSMenuItem alloc] initWithTitle:title action:nil keyEquivalent:@""] autorelease]];
+    NSMenuItem *submenuItem = [[[NSMenuItem alloc] initWithTitle:title action:nil keyEquivalent:@""] autorelease];
```
We need to ARCify MiniBrowser code.

hortont424 replied:
Yeah, may do that at some point.

hortont424 committed:
Simplify MiniBrowser settings menu construction
https://bugs.webkit.org/show_bug.cgi?id=247305

Reviewed by Simon Fraser.

* Tools/MiniBrowser/mac/SettingsController.m:
(-[SettingsController initWithMenu:]):
(addItemToMenuWithTarget):
(addSeparatorToMenu):
(addSubmenuToMenu):
(-[SettingsController _populateMenu:]):
(-[SettingsController buildUserAgentsMenu:]):
(-[SettingsController validateMenuItem:]):
(-[SettingsController changeCustomUserAgent:]):
(-[SettingsController dealloc]): Deleted.
(-[SettingsController _addItemWithTitle:action:indented:]): Deleted.
(-[SettingsController _addHeaderWithTitle:]): Deleted.
(-[SettingsController _populateMenu]): Deleted.
(-[SettingsController changeCutomUserAgent:]): Deleted.
Simplify the menu construction with some more helper functions.
Move the resource usage overlays into the overlay submenu.
Remove the empty WebKit1-only settings section.
Fix some typos.

Canonical link: https://commits.webkit.org/256195@main
